Dynamic quantized nn.LSTM module

jamesr66a opened the pull request:
Previously we only had a Python wrapper for `torch.quantized_lstm_cell`. We had the op `torch.quantized_lstm`, but it didn't have a wrapper. This makes the wrapper

driazati commented on torch/jit/quantized.py:
```diff
@@ -108,10 +110,10 @@ def check_forward_hidden(self, input, hx, hidden_label=''):
                     hidden_label, hx.size(1), self.hidden_size))
 
-    # TODO: for some reason weak_script_method causes a destruction of the
+    # TODO: for some reason torch.jit.script_method causes a destruction of the
```
comments got messed up around here

driazati commented on torch/jit/quantized.py:
```diff
@@ -204,6 +206,234 @@ def forward(self, input, hx=None):
+    # type: (Tensor, Tensor, int) -> Tensor
+    return tensor.index_select(dim, permutation)
+
+class QuantizedRNNBase(torch.jit.ScriptModule):
```
Why did you have to copy all of `nn.LSTM` instead of inheriting from it and adding the `_pack` and `_unpack` methods?

jamesr66a replied:
It's not just `_pack` and `_unpack`, this has a totally new constructor and different attributes. Basically the only things that could be inherited are some of the auxiliary helper methods, which I didn't think were valuable enough

driazati commented on test/test_jit.py:
```diff
@@ -6824,6 +6824,90 @@ def forward(self, x, hiddens):
+
+        # Compare export/import to unquantized
+        export_import_wrapper = self.getExportImportCopyWithPacking(wrapper)
+        ei_out, ei_hid = wrapper(x, hiddens)
```
shouldn't this be `export_import_wrapper(x, hiddens)`?

Commit:
Dynamic quantized full LSTM module (#20249)

Summary:
Previously we only had a Python wrapper for `torch.quantized_lstm_cell`. We had the op `torch.quantized_lstm`, but it didn't have a wrapper. This makes the wrapper
Pull Request resolved: https://github.com/pytorch/pytorch/pull/20249

Reviewed By: driazati

Differential Revision: D15250023

Pulled By: driazati

fbshipit-source-id: 788d8db494f5546abda05751a944d8fb48ea2f62
